patches/server/0807-Discard-friction-API.patch

```diff
@@ -0,0 +1,72 @@
+index 063f3e4c67e6716c9a03dbe4b72eafd32e4f0d53..f3f6b635a056d3014ca6c78e145e12b6e133bb42 100644
+--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
++++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+@@ -53,6 +53,7 @@ public class ItemEntity extends Entity {
+     public final float bobOffs;
+     private int lastTick = MinecraftServer.currentTick - 1; // CraftBukkit
+     public boolean canMobPickup = true; // Paper
++    public boolean discardFriction = false; // Paper
+ 
+     public ItemEntity(EntityType<? extends ItemEntity> type, Level world) {
+         super(type, world);
+@@ -139,7 +140,11 @@ public class ItemEntity extends Entity {
+                 this.move(MoverType.SELF, this.getDeltaMovement());
+                 float f1 = 0.98F;
+ 
+-                if (this.onGround) {
++                // Paper start
++                if (this.discardFriction) {
++                    f1 = 1F;
++                } else if (this.onGround) {
++                    // Paper end
+                     f1 = this.level.getBlockState(new BlockPos(this.getX(), this.getY() - 1.0D, this.getZ())).getBlock().getFriction() * 0.98F;
+                 }
+ 
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+index 342345eb04d00efb58392ccf209e3c51c1064173..410687a651b85bf3931621581d79789bf0ec7735 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+@@ -85,6 +85,16 @@ public class CraftItem extends CraftEntity implements Item {
+      public void setWillAge(boolean willAge) {
+         item.age = willAge ? 0 : NO_AGE_TIME;
+      }
++
++     @Override
++     public boolean discardsFriction() {
++        return this.item.discardFriction;
++     }
++
++     @Override
++     public void setDiscardFriction(boolean discardsFriction) {
++        this.item.discardFriction = discardsFriction;
++     }
+     // Paper End
+ 
+     @Override
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+index fe71123dc07cb13ffad8f13e57aa9bda1cb0abf5..1cb9d793c2f280d8396e4c1e53d8bf86dc4c9c79 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+@@ -915,5 +915,15 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+         }
+         throw new IllegalArgumentException(entityCategory + " is an unrecognized entity category");
+     }
++
++    @Override
++    public boolean discardsFriction() {
++        return getHandle().shouldDiscardFriction();
++    }
++
++    @Override
++    public void setDiscardFriction(boolean discardsFriction) {
++        getHandle().setDiscardFriction(discardsFriction);
++    }
```

Comment:
Several concerns with this relating to LivingEntity, its not persistent, as it not stored in the entities NBT, but also, other parts of the server do change the discardFriction boolean on LivingEntity, specifically relating to some jump behaviors.
I think, instead of hijacking the already existing boolean, a new TriState should be added, ignore, always, and unset to fallback on default behavior